mempool/clist_mempool.go

```diff
@@ -753,3 +766,17 @@ func (rc *recheck) findNextEntryMatching(tx *types.Tx) bool {
 func (rc *recheck) doneRechecking() <-chan struct{} {
 	return rc.doneCh
 }
+
+// setRecheckFull sets recheckFull to true if rechecking is still in progress. It returns true iff
+// the value of recheckFull has changed.
+func (rc *recheck) setRecheckFull() bool {
+	rechecking := !rc.done()
+	recheckFull := rc.recheckFull.Swap(rechecking)
+	return rechecking != recheckFull
+}
+
+// consideredFull returns true iff the mempool should be considered as full while rechecking is in
+// progress.
+func (rc *recheck) consideredFull() bool {
+	return rc.recheckFull.Load()
```

Comment:
I was going to comment that it would be interesting to also check if the rechecking is happening here, but from the `setRecheckFull()` logic seems that the `recheckFull` will be only true when it's is rechecking (`!rc.done()`) and that is covered in that logic I believe